starlette/requests.py

```diff
@@ -239,7 +239,7 @@ async def json(self) -> typing.Any:
             self._json = json.loads(body)
         return self._json
 
-    async def form(self) -> FormData:
+    async def form(self, max_file_size_in_memory: int = 1024 * 1024) -> FormData:
```

Comment:
I'm open to better names for this argument

Reply:
I can do that.

I did a bit of digging (specifically #582). It seems like the max_size attribute is exposed publically but is not customizable by users (other than monkey patching it). So maybe what we can do here is put it in FormData for now, but take advantage of the fact that this will be a breaking change to make it private. Then we can discuss if it makes sense to give users a more direct way of changing the value (e.g. the parameter proposed in this PR or just making it public).

Reply:
> This PR could switch it to an attribute on the FormParser class.

*Oops* - This should have read "an attribute on the `MultiPartParser` class".

Reply:
Hmm ya I like that more!

Reply:
Alright this is done @tomchristie, as always thank you for the great feedback!